### ISSUE TITLE
Dynamo throws Invalid FilterExpression error on a collection query due to redundant parentheses

### ISSUE THREAD

**Issue by Euclidean-Dev** [tywalch/electrodb#239]
**Describe the bug**
When querying a collection, the dynamodb client throws an error concerning the filter expression having more parentheses than needed.  As a workaround, by passing in an edited filter expression in the params object, the query succeeds. 

i.e.
```
 .go({
	params: {
		FilterExpression: "(#__edb_e__ = :__edb_e___client AND #__edb_v__ = :__edb_v___client) OR (#__edb_e__ = :__edb_e___session AND #__edb_v__ = :__edb_v___session)"
	}
})
```

Note that v3 of the AWS SDK is being used:

```
export * as Dynamo from "./dynamo";

import { type EntityConfiguration } from "electrodb";
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
import { Table } from "sst/node/table";

const Client = new DynamoDBClient({ region: "us-east-1" });
export const DocumentClient = DynamoDBDocumentClient.from(Client, {
	marshallOptions: {
		convertEmptyValues: true
	}
});

export const Configuration: EntityConfiguration = {
	table: Table.table.tableName,
	client: DocumentClient
};
```

**AWS SDK Version**
3.319.0

**ElectroDB Version**
2.5.1

**Expected behavior**
The generated filterExpression shouldn't have the outer set of parentheses.

**Errors**
```
{
  KeyConditionExpression: '#pk = :pk and begins_with(#sk1, :sk1)',
  TableName: 'table',
  ExpressionAttributeNames: {
    '#pk': 'pk',
    '#sk1': 'sk',
    '#__edb_e__': '__edb_e__',
    '#__edb_v__': '__edb_v__'
  },
  ExpressionAttributeValues: {
    ':pk': '$app#practitionerid_123',
    ':sk1': '$bookings',
    ':__edb_e___client': 'Client',
    ':__edb_v___client': '1',
    ':__edb_e___session': 'Session',
    ':__edb_v___session': '1'
  },
  FilterExpression: '((#__edb_e__ = :__edb_e___client AND #__edb_v__ = :__edb_v___client) OR (#__edb_e__ = :__edb_e___session AND #__edb_v__ = :__edb_v___session))'
}

ElectroError: Error thrown by DynamoDB client: "Invalid FilterExpression: The expression has redundant parentheses;" -
```

**Comment by tywalch** [comment]
Hey @Euclidean-Dev 👋 

Are you able to share your model and the query that recreates this?

**Comment by Euclidean-Dev** [comment]
Hey @tywalch !  Yep, here's the [playground link](https://electrodb.fun/#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBlTADcsAY1IBfbhiiSwAIkRwjVSzABGNgNzijUPrgVUGFwRSAF5bTShUDAB9AKC4aL4GSTgPLx8-MABhBBx1NQ0qTTAwvjgWVXUtAApRAEhFerrJWEQALiUmuv1MXCwfDpsARht8Lv5CzUGcvKpRruIMQxNBhmlpGybTMbq6hionLBdUKjwOxt26o1yJgEkYc666ouk4Qb9sPgBzecu68gYMAA8nwEFMwE50DtLsBgPAuAxKFQOtUAJQlAB8YFQqCwMH0ABY0V1tl1pBgGI52P0yhh7o8-i83rYPlhvr9LuQ0FhyA8IRgoU8AcDQeDIXASdC6rCwAwBGAWsx9Ex+qghPtDsdTrgukYAacYABBZGdRmaV6DPioSQuTAc3bCkFgjriqV1YgmtGYsAAEX2cAAdHwoCxiZdSZdUNIYP6jSaLpcmZbrbaMPb-nBAU6xQK4G6WPsjHRBiB0x6UeiQli-acgyGw7tTFtoWz4AAPM6m3bkqRMcEJ3beBDBKkDWwuKBQADWbK+uHT0inDL+XBwCD5NkX6auVlkfVOHQA2puKVSND5MPcbABdJ4Ry64Jddy6rxAbx-b7wyKD75nH66zFet5-E24ZSi4mgAAqnoUNKYMug5QMODjnnwgzeOolI0OmrZwG2gxzlgIxuouCG7K+64EX0Qxbm6O7fr+R4nlh1IXnSMA3nebqPmRdQURuhFDB+dFfnu7B-jYeqZgaxqcSBkpdBBMwTGRuH4bYhEAEwLk+A7kWuAl9JptFPPRYkHmA-43OoQFcU8PHPvpb5UVgmnCaZok-uJTFSbGsnAeGJJbPUEaKHEwRNqiniiBhmR6LgfQ+AUWglGAZQVMlRS1A0TQtPACAIT0GCJWhtjEeMVRFIM8UlfaizLMyNhrBszZNBq2Bap2A7EAlNL0o5SYsoc7Juo6oourmbqwvCiLUBW3o4nihINnU951OSLGoZefJ6YNNisiNQpwNyvITYKfxjc6-LoAplwAXcO1PHtB0-KNx24qd115kdWbjV9t11Lq+pwHGZF7VaNp2m9v1Xa69lwJ6lbVv6dahqiAN1FGMYyfGT3mo1EOpuml05udlwFlQRYlmWCPzVWvoo8GaNBY2LYCHhXVktgkh9mRQ4jqhgwTtOs7ziRummfxgwmX8ZleRZx4bWecHsXJYH2RLstSyyU6fru8sSfdNkcYQ+14CVtnyerlxKdZuN-GpLnaeLvHazYWky38nmMZZkl25b1sPprK4GU77my973m+75ONq42LOrSFOxhYEEWiFF6S+AoynqDVNJ6EsxihGl5RgAXDXZQmRsmjnVCZdoTQ9SVHR50llWaEn9Qp-EkXRTKKgYJYGAQnQlgsHwYAwOgEJQGAkSnMP0hMBMdB4Hgoi163fDl0XAb8yhNI6rLdQBsLM7fLg1R6ZcSuwWx-XDJpADMmzH6Y6PHwGXxQGigMykCRVsDwFYHIKcOpN7m3zgYXe+9Ry+FMifM+osr4IPWjBVitIH5DGfq-WW78EFfx-tfbsTBki4F4pcAAYlgagmAVBtnJJAgYqDdg2GqAAYmiNEEGLhuFcNSm0LhPC+FcOrmAQ0AA5H0YBOHcNcNEfQ-CwiCLkbwxRoi7boiBAAJTABwoR8iEjRAEQY3hRjohNxpOIqRMjTEKKUWAFRwj1EWKYXwVEuC8EIPfkAA)...

**Comment by tywalch** [comment]
@Euclidean-Dev are you able to put together an example where DynamoDB throws? I recreated your example and executed it successfully (the client did not throw an exception) with both the v2 client and v3 client here: 

https://gist.github.com/tywalch/6ca99d717189af78591f7308e8658283

**Comment by Euclidean-Dev** [comment]
Hi @tywalch.  

Hmmm. This is very strange. 

So I just now tried a couple of things to see whether I could narrow it down some more.  And noticed that it only throws the redundant parentheses error if I  do the following:

```
console.log(query.params());
const clients = await query.go();
```


But it doesn't throw if I comment out the console.log?  
```
//console.log(query.params());
const clients = await query.go();
```

**Comment by tywalch** [comment]
This is because you're calling a terminal operation ("go" and "params") twice on the same chain "query"

**Comment by Euclidean-Dev** [comment]
Ah! Sorry! Of course it is, the error response confused me and I totally missed that. Thanks @tywalch

**Comment by tywalch** [comment]
This ticket is good though because I've seen someone else do this once as well, so I think there are improvements that could be made here 👍

**Comment by Euclidean-Dev** [comment]
Cool! I don't want to waste any of your time. You're a legend! I've not looked at electrodb's source yet but maybe even something simple to prompt as a reminder could be good for us plebs.  😂

```
    if (this.terminalOperationCalled) {
      throw new Error("Terminal operation (go or params) has already been called on this query chain.");
    }
```

**Comment by tywalch** [comment]
@Euclidean-Dev 
A fix for this is in ElectroDB version `2.6.0` 👍

**Comment by Euclidean-Dev** [comment]
Thanks @tywalch !!  You da man!
